iceoryx2-bb/container/src/queue.rs

```diff
@@ -10,39 +10,74 @@
 //
 // SPDX-License-Identifier: Apache-2.0 OR MIT
 
-//! Two relocatable (inter process shared memory compatible) queues.
+//! Three queue variations that are similar to [`std::collections::VecDeque`].
 //!
-//! The [`Queue`] which has a
-//! fixed capacity defined at runtime and the [`FixedSizeQueue`] which has a fixed capacity at
-//! compile time.
+//!  * [`FixedSizeQueue`](crate::queue::FixedSizeQueue), compile-time fixed size queue that
+//!     is self-contained.
+//!  * [`RelocatableQueue`](crate::queue::RelocatableQueue), run-time fixed size queue that
+//!     acquires the required memory from a custom user-provided allocator.
+//!  * [`Queue`](crate::queue::Queue), run-time fixed size queue that uses by default
+//!     heap memory.
 //!
-//! # Examples
+//! # User Examples
 //!
-//! ## Create [`Queue`] inside constructs which provides memory
+//! ## Use the [`FixedSizeQueue`](crate::queue::FixedSizeQueue)
+//!
+//! ```
+//! use iceoryx2_bb_container::queue::FixedSizeQueue;
+//!
+//! const QUEUE_CAPACITY: usize = 1;
+//! let mut queue = FixedSizeQueue::<u64, QUEUE_CAPACITY>::new();
+//!
+//! queue.push(123);
+//!
+//! // queue is full, we override the oldest element (123) with the new number (456)
+//! queue.push_with_overflow(456);
+//!
+//! println!("pop from queue {}", queue.pop().unwrap());
+//! ```
+//!
+//! ## Use the [`Queue`](crate::queue::Queue)
+//!
+//! ```
+//! use iceoryx2_bb_container::queue::Queue;
+//!
+//! let queue_capacity = 1234;
+//! let mut queue = Queue::<u64>::new(queue_capacity);
+//!
+//! queue.push(123);
+//!
+//! println!("pop from queue {}", queue.pop().unwrap());
+//! ```
+//!
+//! # Expert Examples
+//!
+//! ## Create [`RelocatableQueue`](crate::queue::RelocatableQueue) inside constructs which provides memory
 //!
 //! ```
 //! use iceoryx2_bb_container::queue::RelocatableQueue;
 //! use iceoryx2_bb_elementary::math::align_to;
 //! use iceoryx2_bb_elementary::relocatable_container::RelocatableContainer;
+//! use core::mem::MaybeUninit;
 //!
 //! const QUEUE_CAPACITY:usize = 12;
 //! struct MyConstruct {
 //!     queue: RelocatableQueue<u128>,
-//!     data: [u128; QUEUE_CAPACITY],
+//!     queue_memory: [MaybeUninit<u128>; QUEUE_CAPACITY],
 //! }
 //!
 //! impl MyConstruct {
 //!     pub fn new() -> Self {
 //!         Self {
 //!             queue: unsafe { RelocatableQueue::new(QUEUE_CAPACITY,
-//!                             align_to::<u128>(std::mem::size_of::<u128>()) as isize) },
-//!             data: [0; QUEUE_CAPACITY]
+//!                             align_to::<MaybeUninit<u128>>(std::mem::size_of::<RelocatableQueue<u128>>()) as isize) },
```

Comment:
Out of curiosity since I'm lazy about writing, does clippy generate a warning when using just `align_to::<u128>`? Since `MaybeUnit` is transparent, it should be equivalent

Reply:
No, it does not but I wanted to do this by the book.